lib/plugins/aws/provider.js

```diff
@@ -1106,8 +1106,13 @@ class AwsProvider {
             },
             vpc: { $ref: '#/definitions/awsLambdaVpcConfig' },
             vpcEndpointIds: {
-              type: 'array',
-              items: { $ref: '#/definitions/awsCfInstruction' },
+              anyOf: [
+                {
+                  type: 'array',
+                  items: { $ref: '#/definitions/awsCfInstruction' },
+                },
+                { $ref: '#/definitions/awsCfSplit' },
```

Comment:
Why not include `awsCfSplit` as part of AWS CF instructions set?

Reply:
I was thinking about it, but to be honest, `awsCfInstruction` is a little bit different as in most cases the supported functions resolve to a single identifier, where the `Fn::Split` produces an array. For example, in this case we don't want to accept `Fn::Split` as an item in array because that would result in invalid value for the `VpcEndpointIds` property

Reply:
I see, yes it makes sense then to not include it.

But now I wonder whether it'll not be wise to introduce something as `awsCfArrayInstruction` that will be an alias to above.

As that construct feels pretty generic and having that, we may agree to use it for every array property where we allow CF intrinsic functions.

What do you think?

Reply:
On a second though, the current `awsCfInstruction` might be a bit misleading as it doesn't include `Fn::Split`, which is also a `cfInstruction`, but it's often not supported in the same places as the current definitions in `cfInstruction` - maybe we could rename the `awsCfInstruction` to `awsCfInstructionSingleResult` to be more explicit?

Reply:
Sorry, I was writing the above comment without seeing your response - I like the proposed idea :+1: I think it will be useful as I think we might have a few other places where we don't support `Fn::Split` by omission in schema